src/BaseIri.js

```
@@ -0,0 +1,132 @@
+export default class BaseIri {
+  constructor(iri) {
+    if (iri.startsWith('file://')) {
+      throw new Error('Base IRIs starting with file:// are not supported.');
+    }
+
+    // Generate regex for baseIRI with optional groups for segments
+    // Stage 0: find the first scheme delimiter -> stage 1
+    // Stage 1: find the next /, ?, or #. '/' -> new segment -> stage 1, '?' -> stage 2, '#' -> update end to position before hash -> stage 3, none -> stage 3
+    // Stage 2: find the next #. '#' -> update end to position before hash -> stage 3, none -> stage 3
+    // Stage 3: find the end of the string, add '$' to this last segment
+    this._baseSubstitutions = {};
+    let baseIRIRegex = '';
+    let segmentsCount = 0;
+    let stage = 0;
+    const slashPositions = [];
+    let i = 0;
+    let containsQuery = false;
+    // Stage 0
+    const match = /:\/{0,2}/.exec(iri);
+    if (match) {
+      baseIRIRegex += escapeRegex(iri.substring(0, match.index + match[0].length));
+      i = match.index + match[0].length;
+      stage = 1;
+    }
+    else {
+      throw new Error(`Base IRI should contain a scheme followed by its delimiter (e.g., http://). Instead, found: '${iri}'.`);
+    }
+
+    if (/\/\.{0,2}\//.test(iri.substring(i))) {
+      throw new Error('Base IRIs containing `//`, `/./`, or `/../` are not supported.');
```

Perhaps also silent failure here is okay.